backend/tournesol/serializers/rating.py

```diff
@@ -51,24 +51,24 @@ def to_internal_value(self, data):
 
 
 class ContributorRatingCreateSerializer(ContributorRatingSerializer):
-    video_id = RegexField(YOUTUBE_VIDEO_ID_REGEX, write_only=True)
+    uid = RegexField(rf"yt:{YOUTUBE_VIDEO_ID_REGEX[1:]}", write_only=True)
 
     class Meta:
         model = ContributorRating
-        fields = ["video_id", "is_public", "video", "criteria_scores", "n_comparisons"]
+        fields = ["uid", "is_public", "entity", "criteria_scores", "n_comparisons"]
 
     def validate(self, attrs):
-        video_id = attrs.pop("video_id")
-        video_serializer = RelatedVideoSerializer(data={"video_id": video_id})
-        video_serializer.is_valid(raise_exception=True)
-        video = Entity.get_from_video_id(video_id)
+        uid = attrs.pop("uid")
+        entity_serializer = RelatedEntitySerializer(data={"uid": uid})
+        entity_serializer.is_valid(raise_exception=True)
+        entity = Entity.objects.get(uid=uid)
         user = self.context["request"].user
-        if user.contributorvideoratings.filter(entity=video).exists():
+        if user.contributorvideoratings.filter(entity=entity).exists():
             raise ValidationError(
-                "A ContributorRating already exists for this (user, video)",
+                "A ContributorRating already exists for this (user, entity)",
                 code="unique",
             )
```

Comment:
The uniqueness should be checked for (user, entity, poll) instead of (user, entity)

Reply:
That's true, it also means a test is missing here.